tests/test_core.py

```diff
@@ -254,42 +243,37 @@ def test_handle_error_reraises_errors(web_request):
         p.handle_error(ValidationError("error raised"), web_request, Schema())
 
 
-@mock.patch("webargs.core.Parser.parse_headers")
-def test_locations_as_init_arguments(parse_headers, web_request):
-    p = Parser(locations=("headers",))
+@mock.patch("webargs.core.Parser.location_load_headers")
+def test_location_as_init_argument(location_load_headers, web_request):
+    p = Parser(location="headers")
+    location_load_headers.return_value = {}
     p.parse({"foo": fields.Field()}, web_request)
-    assert parse_headers.called
+    assert location_load_headers.called
 
 
-@mock.patch("webargs.core.Parser.parse_files")
-def test_parse_files(parse_files, web_request):
-    p = Parser()
-    p.parse({"foo": fields.Field()}, web_request, locations=("files",))
-    assert parse_files.called
-
-
-@mock.patch("webargs.core.Parser.parse_json")
-def test_custom_error_handler(parse_json, web_request):
+def test_custom_error_handler(web_request):
     class CustomError(Exception):
         pass
 
     def error_handler(error, req, schema, status_code, headers):
         assert isinstance(schema, Schema)
         raise CustomError(error)
 
-    parse_json.side_effect = ValidationError("parse_json failed")
+    mock_schema = mock.Mock(spec=Schema)
+    mock_schema.strict = True
+    mock_schema.load.side_effect = ValidationError("parsing json failed")
```

Comment:
Is the mocking here necessary?

Reply:
This is a good catch. I'd rather use a real schema object where possible. Will fix.